draft-bbs-signatures.md

```diff
@@ -389,15 +389,14 @@ Procedure:
 This operation computes a deterministic signature from a secret key (SK) and optionally over a header and or a vector of messages.
 
 ```
-signature = Sign(SK, PK, header, (msg_1,..., msg_L), (H_1,..., H_L))
+signature = Sign(SK, PK, header, [(H_1, msg_1),..., (H_L, msg_L)])
 
 Inputs:
 
 - SK (REQUIRED), an octet string of the form outputted by the KeyGen operation.
 - PK (REQUIRED), an octet string of the form outputted by the SkToPk operation provided the above SK as input.
 - header (OPTIONAL), an octet string containing context and application specific information. If not supplied, it defaults to an empty string.
-- msg_1,..., msg_L (OPTIONAL), a vector of octet strings.
-- H_1,..., H_L (OPTIONAL), points of G1. Generators used to commit each message.
+- [(H_1, msg_1),..., (H_L, msg_L)] (OPTIONAL), a list with pairs of the form (point in G1, scalar). Generators and messages pairs, corresponding to messages to be signed. If not supplied, it will default to the empty array ("[]"). The list MUST be ordered based on the values of the generators  (i.e., H_1 < H_2 < ... < H_L).
```

Comment:
The list MUST be ordered based on the index of the generators?

Reply:
This does make sense considering how the generators are created but that order is harder to be "verified" unless the verifier re-creates all the generators (not really a security issue I think, it's just that the proof will not validate). 

Used the suggestion for now and opened issue #141 to discuss more.